common/lib/xmodule/xmodule/modulestore/split_mongo/mongo_connection.py

```diff
@@ -203,6 +209,40 @@ def structure_to_mongo(structure, course_context=None):
         return new_structure
 
 
+class CourseStructureCache(object):
+    """
+    Wrapper around django cache object to cache course structure objects.
+    The course structures are pickled and compressed when cached.
+    """
+    def __init__(self):
+        try:
+            self.cache = get_cache('course_structure_cache')
+        except InvalidCacheBackendError:
+            self.cache = get_cache('default')
```

Comment:
My understanding is that it's LRU by last access, so by definition whatever is getting evicted would be used less often than this. The biggest danger I can think of is this stomping on sessions for inactive users. It's a good argument for isolating the session caches from the content pieces, but I don't think changing the expiration time helps much.